staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/aes/aes_cbc_hmac.go

```diff
@@ -0,0 +1,217 @@
+package aes
+
+import (
+	"bytes"
+	"crypto/cipher"
+	"crypto/hmac"
+	"crypto/rand"
+	"errors"
+	"hash"
+	"io"
+)
+
+// AEAD is an interface that mimics cipher.AEAD to implement an
+// Authenticated Encryption with Associated Data scheme.
+// This implements the AEAD scheme as defined in RFC5116 at:
+// https://tools.ietf.org/html/rfc5116
+type AEAD interface {
```

Comment:
I don't think you need to create this interface again here, the return type in your funcs below can be cipher.AEAD since go uses duck typing

Reply:
Yes, this is true and I considered using cipher.AEAD directly. However, I wanted someone who inspected the code diff (e.g. Stefano) to readily have the interface viewable so that they know what the interface dictates and which methods need to be implemented. A good alternative would be to use cipher.AEAD and lay out the interface implementation in the report.

Reply:
alternative sounds good